Singularity too old

The grpc4bmi library requires Singularity 3.1.0, while 3.0.3 is installed.

We should update the singularity version.

Ran `ansible-playbook -i inventory.yml jupyter.yml` with all tasks except singularity role commented out. 

This gave:
```
TASK [singularity : Make] ********************************************************************************************************************************************************************
fatal: [jupyter.ewatercycle.org]: FAILED! => {"changed": false, "cmd": "/usr/bin/make", "msg": "\n========================================================================\nA non-backwards compatible change has been added to the build system and\nit's necessary to start fresh.\n\nPlease remove the build directory (\"builddir\" by default) and run\nmconfig again.\n========================================================================\n\n\n========================================================================\nA non-backwards compatible change has been added to the build system and\nit's necessary to start fresh.\n\nPlease remove the build directory (\"builddir\" by default) and run\nmconfig again.\n========================================================================\n\nexit status 1\nmake: *** [singularity.conf] Error 1", "rc": 2, "stderr": "\n========================================================================\nA non-backwards compatible change has been added to the build system and\nit's necessary to start fresh.\n\nPlease remove the build directory (\"builddir\" by default) and run\nmconfig again.\n========================================================================\n\n\n========================================================================\nA non-backwards compatible change has been added to the build system and\nit's necessary to start fresh.\n\nPlease remove the build directory (\"builddir\" by default) and run\nmconfig again.\n========================================================================\n\nexit status 1\nmake: *** [singularity.conf] Error 1\n", "stderr_lines": ["", "========================================================================", "A non-backwards compatible change has been added to the build system and", "it's necessary to start fresh.", "", "Please remove the build directory (\"builddir\" by default) and run", "mconfig again.", "========================================================================", "", "", "========================================================================", "A non-backwards compatible change has been added to the build system and", "it's necessary to start fresh.", "", "Please remove the build directory (\"builddir\" by default) and run", "mconfig again.", "========================================================================", "", "exit status 1", "make: *** [singularity.conf] Error 1"], "stdout": " CSCOPE\n DEPENDS\n GEN singularity.conf from /etc/singularity/singularity.conf\nUsage: go run ... [infile] <outfile>\nMakefile:352: recipe for target 'singularity.conf' failed\n", "stdout_lines": [" CSCOPE", " DEPENDS", " GEN singularity.conf from /etc/singularity/singularity.conf", "Usage: go run ... [infile] <outfile>", "Makefile:352: recipe for target 'singularity.conf' failed"]}
```

--- a/roles/singularity/README.md
+++ b/roles/singularity/README.md
@@ -17,6 +17,7 @@ Role Variables
 | singularity_version | Singularity version |
 | singularity_sha256sum | sha256 checksum of Singularity release tarball |
 | go_version | Go version |
+| go_sha256sum | sha256 checksum of Go binary tarball |
 
 Example Playbook
 ----------------

--- a/roles/singularity/defaults/main.yml
+++ b/roles/singularity/defaults/main.yml
@@ -1,5 +1,5 @@
 ---
-singularity_version: '3.0.3'
-singularity_sha256sum: 2476df2fa2ef3488a2cf8fc42ed0dc3536df09cd1ca40fa7febc086ed677a047
+singularity_version: '3.5.2'
+singularity_sha256sum: f9c21e289377a4c40ed7a78a0c95e1ff416dec202ed49a6c616dd2c37700eab8
 go_version: '1.11.5'
 go_sha256sum: ''

--- a/roles/singularity/defaults/main.yml
+++ b/roles/singularity/defaults/main.yml
@@ -2,4 +2,4 @@
 singularity_version: '3.5.2'
 singularity_sha256sum: f9c21e289377a4c40ed7a78a0c95e1ff416dec202ed49a6c616dd2c37700eab8
 go_version: '1.11.5'
-go_sha256sum: ''
+go_sha256sum: 'ff54aafedff961eb94792487e827515da683d61a5f9482f668008832631e5d25'

--- a/roles/singularity/defaults/main.yml
+++ b/roles/singularity/defaults/main.yml
@@ -1,5 +1,5 @@
 ---
 singularity_version: '3.5.2'
 singularity_sha256sum: f9c21e289377a4c40ed7a78a0c95e1ff416dec202ed49a6c616dd2c37700eab8
-go_version: '1.11.5'
-go_sha256sum: 'ff54aafedff961eb94792487e827515da683d61a5f9482f668008832631e5d25'
+go_version: '1.13.8'
+go_sha256sum: '0567734d558aef19112f2b2873caa0c600f1b4a5827930eb5a7f35235219e9d8'

--- a/roles/singularity/tasks/build.yml
+++ b/roles/singularity/tasks/build.yml
@@ -8,17 +8,17 @@
     url: https://github.com/sylabs/singularity/releases/download/v{{ singularity_version }}/singularity-{{ singularity_version }}.tar.gz
     dest: /mnt/apps
     checksum: sha256:{{ singularity_sha256sum }}
+  register: sing_fetch_result
+- name: Clear singularity dir
+  file:
+    path: /root/go/src/github.com/sylabs/singularity
+    state: absent
+  when: sing_fetch_result.changed
 - name: Unpack singularity tarball
   unarchive:
     src: /mnt/apps/singularity-{{ singularity_version }}.tar.gz
     dest: /root/go/src/github.com/sylabs
     remote_src: yes
-  register: unpack_result
-- name: Clear builddir
-  file:
-    path: /root/go/src/github.com/sylabs/singularity/builddir
-    state: absent
-  when: unpack_result.changed
 - name: Go dependencies
   command: go get -u -v github.com/golang/dep/cmd/dep
   args:

--- a/roles/singularity/tasks/build.yml
+++ b/roles/singularity/tasks/build.yml
@@ -13,6 +13,12 @@
     src: /mnt/apps/singularity-{{ singularity_version }}.tar.gz
     dest: /root/go/src/github.com/sylabs
     remote_src: yes
+  register: unpack_result
+- name: Clear builddir
+  file:
+    path: /root/go/src/github.com/sylabs/singularity/builddir
+    state: absent
+  when: unpack_result.changed
 - name: Go dependencies
   command: go get -u -v github.com/golang/dep/cmd/dep
   args:

--- a/roles/singularity/tasks/go.yml
+++ b/roles/singularity/tasks/go.yml
@@ -4,6 +4,12 @@
     url: https://dl.google.com/go/go{{ go_version }}.linux-amd64.tar.gz
     dest: /mnt/apps
     checksum: sha256:{{ go_sha256sum }}
+  register: fetch_go_result
+- name: Drop go dir
+  file:
+    path: /usr/local/go
+    state: absent
+  when: fetch_go_result.changed
 - name: Unarchive Go binary tarball
   unarchive:
     src: /mnt/apps/go{{ go_version }}.linux-amd64.tar.gz

--- a/roles/singularity/tasks/go.yml
+++ b/roles/singularity/tasks/go.yml
@@ -3,7 +3,7 @@
   get_url:
     url: https://dl.google.com/go/go{{ go_version }}.linux-amd64.tar.gz
     dest: /mnt/apps
-    checksum: sha256:ff54aafedff961eb94792487e827515da683d61a5f9482f668008832631e5d25
+    checksum: sha256:{{ go_sha256sum }}
 - name: Unarchive Go binary tarball
   unarchive:
     src: /mnt/apps/go{{ go_version }}.linux-amd64.tar.gz

--- a/roles/singularity/tasks/main.yml
+++ b/roles/singularity/tasks/main.yml
@@ -8,6 +8,7 @@
     - uuid-dev
     - libgpgme11-dev
     - squashfs-tools
+    - cryptsetup-bin
 - name: Get Go
   include: go.yml
 - name: Build singularity

